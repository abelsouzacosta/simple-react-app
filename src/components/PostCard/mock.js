const mock = {
  title: "Titulo 1",
  body: "body1",
  cover: "img/img.png",
  id: 1,
};

export default mock;
