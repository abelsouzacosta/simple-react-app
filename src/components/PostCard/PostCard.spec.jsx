import { render, screen } from "@testing-library/react";
import { PostCard } from ".";
import mock from "./mock";

describe("<PostCard />", () => {
  it("should renderized correctly", () => {
    render(<PostCard {...mock} />);

    expect(screen.getByRole("img", { alt: "title 1" })).toHaveAttribute(
      "src",
      mock.cover
    );

    expect(
      screen.getByRole("heading", { title: "title 1 1" })
    ).toBeInTheDocument();

    expect(screen.getByText("body1")).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = render(<PostCard {...mock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
