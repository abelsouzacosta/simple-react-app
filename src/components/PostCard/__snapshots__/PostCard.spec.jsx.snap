// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Titulo 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h3>
      Titulo 1
       
      1
    </h3>
    <p>
      body1
    </p>
  </div>
</div>
`;
